### ISSUE TITLE
Robustness: insecure mode is not working correctly

### ISSUE THREAD

**Issue by PascalinDe** [dedis/dela#271]
```
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]: panic: certificate of the overlay must be populated
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]: goroutine 46 [running]:
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]: go.dedis.ch/dela/mino/minogrpc.(*overlay).GetCertificateChain(0xc000000540)
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]:         /home/runner/go/pkg/mod/go.dedis.ch/dela@v0.0.0-20230907131212-0d61b081b4af/mino/minogrpc/server.go:527 +0x6f
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]: go.dedis.ch/dela/mino/minogrpc/controller.tokenAction.Execute({}, {{0xdb3610, 0xc000042690}, {0xdba740, 0xc000343e60}, {0xdafb20, 0xc000042f28}})
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]:         /home/runner/go/pkg/mod/go.dedis.ch/dela@v0.0.0-20230907131212-0d61b081b4af/mino/minogrpc/controller/actions.go:111 +0x12d
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]: go.dedis.ch/dela/cli/node.(*socketDaemon).handleConn(0xc0000c9340, {0xdbd600?, 0xc000042f18})
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]:         /home/runner/go/pkg/mod/go.dedis.ch/dela@v0.0.0-20230907131212-0d61b081b4af/cli/node/daemon.go:185 +0x58a
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]: created by go.dedis.ch/dela/cli/node.(*socketDaemon).Listen.func2 in goroutine 40
Sep 12 15:23:43 dedis-152.icsil1.epfl.ch start-dvoting[5237]:         /home/runner/go/pkg/mod/go.dedis.ch/dela@v0.0.0-20230907131212-0d61b081b4af/cli/node/daemon.go:129 +0x6c
```

insecure mode keeps checking the certificates anyway

refer to https://github.com/dedis/dela/pull/232/files for details

**Comment by pierluca** [comment]
And not providing the certificates means that the certificate hash (as checked from another node) will not match when doing the `minogrpc join`

**Comment by PascalinDe** [comment]
@pierluca could it be related to this issue as well https://github.com/dedis/dela/issues/220?

**Comment by pierluca** [comment]
This problem seems a regression introduced in #232 and unrelated to #220.
Basically, #220 is about D-Voting not reloading its certificates when the Let's Encrypt certbot updates them.

**Comment by ineiti** [comment]
Closed by c4dt/dela#3
